src/main/java/org/jenkinsci/plugins/gitclient/Git.java

```diff
@@ -116,23 +126,37 @@ public Git using(String exe) {
      * @throws java.lang.InterruptedException if interrupted.
      */
     public GitClient getClient() throws IOException, InterruptedException {
-        jenkins.MasterToSlaveFileCallable<GitClient> callable = new jenkins.MasterToSlaveFileCallable<GitClient>() {
-            public GitClient invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
-                if (listener == null) listener = TaskListener.NULL;
-                if (env == null) env = new EnvVars();
 
-                if (exe == null || JGitTool.MAGIC_EXENAME.equalsIgnoreCase(exe)) {
+        if (env == null) env = new EnvVars();
+        if (repository == null) {
+            // Assume local directory
+            if (launcher != null) {
+                repository = new FilePath(launcher.getChannel(), ".");
+            } else {
+                repository = new FilePath(new File("."));
+            }
+        }
+        GitClient git;
+
+        if (exe == null || JGitTool.MAGIC_EXENAME.equalsIgnoreCase(exe)) {
```

Comment:
@ndeloof Does the exe == null condition here mean  that now the default git implementation on an agent is JGit, whether or not the job selected JGit as the implementation?

If so, that's bad.  Users want their job selected git implementation honored, even if it is the system default.